### ISSUE TITLE
[RFE] Warning when graphroot is on an NFS volume

### ISSUE THREAD

**Issue by jcpunk** [containers/podman#7091]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Can rootless podman emit a warning if 'backingFs=nfs`

**Steps to reproduce the issue:**

1. Build system with /home/ on NFS

2. Setup user namespaces

3. run rootless podman

4. Get error

**Describe the results you received:**

Rootless podman does not run with graphroot on an NFS volume.  No clear hint that it is an Unsupported Filesystem is provided.

**Describe the results you expected:**

Since NFS cannot support rootless podman, could podman emit a notice of some type so that users are aware they need to alter their storage config.

Eventually I found : https://www.redhat.com/sysadmin/rootless-podman-nfs

**Additional information you deem important (e.g. issue happens only occasionally):**

Since this is a known incompatibility, emitting a pointer to the possible issue at some log level would be very helpful.

**Output of `podman version`:**

```
podman version
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.13.4
OS/Arch:            linux/amd64
```

**Comment by mheon** [comment]
@nalind @rhatdan Would you agree this looks like a c/storage feature?

**Comment by nalind** [comment]
Yes, this is something we can detect (we already check for overlay-on-overlay, for example).

**Comment by rhatdan** [comment]
I think we could add a function to check this, but is this something we should report as an error or just allow Podman and Buildah to run the check. After all the issue is User Namespace on top of a NFS share.

**Comment by tin6150** [comment]
Further enhancement/suggestion:

Per https://www.redhat.com/sysadmin/rootless-podman-nfs, 
rootless container on a shared system (and NFS home dir) require each user to fiddle with their ~/.config/containers/storage.conf .  A warning that image storage has defaulted to NFS and things would fail is helpful.  But automagically setting it to another path that works would drive adoption :) 

So, when the system detect it is on NFS share, could it set to use BASEDIR that is not NFS, eg /var/tmp/$UID ? and graphroot = $BASEDIR/containers/storage 

can this settings be done system-wide (or even site-wide), eg to /etc/containers/storage.conf 
This way root has a centralized place to configure it for all users?

Thx!

**Comment by rhatdan** [comment]
We don't do this automatically but there is a place for the admin to configure this data.

```
man storage.conf
...
       rootless_storage_path="$HOME/.local/share/containers/storage"
         Storage  path  for  rootless  users.  By  default  the  graphroot  for  rootless  users  is  set   to
       $XDG_DATA_HOME/containers/storage,  if  XDG_DATA_HOME  is  set.   Otherwise $HOME/.local/share/contain‐
       ers/storage is used.  This field can be used if administrators need to change the storage location  for
       all users.

              The rootless storage path supports three substations:
              * `$HOME` => Replaced by the users home directory.
              * `$UID`  => Replaced by the users UID
              * `$USER` => Replaced by the users name

       A common use case for this field is to provide a local storage directory when user home directories are
       NFS-mounted (podman does not support container storage over NFS).
```
